src/main/java/com/example/bookstore/model/Role.java

```diff
@@ -0,0 +1,40 @@
+package com.example.bookstore.model;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.EnumType;
+import jakarta.persistence.Enumerated;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+import org.springframework.security.core.GrantedAuthority;
+
+@Data
+@Entity
+@SQLDelete(sql = "UPDATE roles SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+@Table(name = "roles")
+public class Role implements GrantedAuthority {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false, unique = true)
+    private RoleName name;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+    @Override
+    public String getAuthority() {
+        return RoleName.USER.name();
```

Comment:
```suggestion
        return "ROLE_" + name.name();
```
Or you may not add `ROLE_`. This may affect your controller method security. There may be a situation when you get 401 because spring wants that prefix, but you don't add.

I recommend to test if it works properly and read about `hasRole` vs `hasAuthrority` if it is not working